src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -697,13 +700,26 @@ private IOperation CreateBoundConversionOperation(BoundConversion boundConversio
                 }
 
                 Lazy<IOperation> operand = new Lazy<IOperation>(() => Create(boundConversion.Operand));
+                ITypeSymbol type = boundConversion.Type;
+                Optional<object> constantValue = ConvertToOptional(boundConversion.ConstantValue);
+
+                // If this is a lambda conversion, we need to check to see if it's a delegate conversion.
+                if (boundConversion.Operand.Kind == BoundKind.Lambda ||
+                    boundConversion.Operand.Kind == BoundKind.UnboundLambda)
+                {
+                    // We need to determine if this is a conversion to a delegate type. If it is, it's a delegate creation, so we
+                    // return an IDelegateCreationExpression instead of a conversion.
+                    if (boundConversion.Type.InheritsSpecialType(SpecialType.System_Delegate))
```

Comment:
>.InheritsSpecialType(SpecialType.System_Delegate) [](start = 44, length = 49)

There is already a helper called IsDelegateType, compiler uses it to detect delegate types.

Reply:
Ah, missed it because it's on TypeSymbol, I was only looking at ITypeSymbol. I'll remove this.